### PR TITLE
Allow displayed name to be more than just first & last

### PR DESCRIPTION
**Pull request by jasongfleischer**
These changes allow both the about page and the header to both display things like middle initials & names, patronymics, or other longer name forms.

### CODE REVIEW

File: _includes/header.html [hunk 1]
```diff
@@ -5,7 +5,9 @@
     <div class="container">
       {% if page.title != "about" %}
         {% assign name = site.title | split: ' ' %}
-        <a class="navbar-brand title font-weight-lighter" href="{{ site.baseurl | prepend: site.url }}/"><span class="font-weight-bold">{{ name | first }}</span> {{ name | last }}</a>
+        {% assign firstname =  name | first %}
```
Comment (alshedivat) [inline _includes/header.html:8]: Could we explicitly have 3 variables, `firstname`, `midlenames`, and `lastname`? If the name has multiple parts, then `firstname` and `lastname` will stand for the first and last words in the name and the `middlenames` will be all the intermediate words.

File: _layouts/about.html [hunk 2]
```diff
@@ -5,9 +5,11 @@
 <div class="post">
 
   <header class="post-header">
-    {% assign name = site.title | split: ' ' %}
-    <h1 class="post-title"><span class="font-weight-bold">{{ name | first }}</span> {{ name | last }}</h1>
-    <p class="post-description">{{ page.description }}</p>
+     {% assign name = site.title | split: ' ' %}
+     {% assign firstname =  name | first %}
```
Comment (alshedivat) [inline _layouts/about.html:9]: Same comment as the above.

**Comment by alshedivat** [comment]
Closing this PR as it seems to be linked to the `master` branch of a personal webpage. Please create a separate clean PR to address the raised issue with "middle initials & names, patronymics, or other longer name forms." Thanks.

**Comment by jasongfleischer** [comment]
Sorry about the messy git'ing.  Will do as requested on a new branch off v0.3.0

**Pull request closed without merging**